constellations/orion/src/controllers/classes_by_semester_controller_test.go

```diff
@@ -0,0 +1,80 @@
+package controllers_test
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/controllers/testUtils"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test getting programs and classes by semester endpoint
+func TestGetClassesAndProgramsBySemester_Success(t *testing.T) {
```

Comment:
I think the best way is to really guarantee this endpoint works is to use a series of test testing different scenarios.
One could be 1 semester, 1 program, 1 class.
Then 1 semester, 1 program, 2 classes
Then 1 semester 2 programs, 2 classes and so on.

Reply:
I think I'll add in the rest of the code after I figure out why it's still returning a 400